File: tutorias-inf-uva-frontend/src/app/models/estado-solicitud.ts
import { EstadoSolicitudEnum } from "./enums/estado-solicitud-enum";

export class EstadoSolicitud {
	id: number;
	estado: string;

	constructor(id: number, estado: string) {
		this.id = id;
		this.estado = estado;
	}

	public static getEstadoSolicitudByEnum(estadoEnum: EstadoSolicitudEnum): EstadoSolicitud {
		switch (estadoEnum) {
			case EstadoSolicitudEnum.ACEPTADA:
				return new EstadoSolicitud(2, 'Aceptada');
			case EstadoSolicitudEnum.RECHAZADA:
				return new EstadoSolicitud(3, 'Rechazada');
			case EstadoSolicitudEnum.CONFIRMADA:
				return new EstadoSolicitud(4, 'Confirmada');
			case EstadoSolicitudEnum.CANCELADA:
				return new EstadoSolicitud(5, 'Cancelada');
			default:
				return new EstadoSolicitud(1, 'Pendiente');
		}
	}
}

File: tutorias-inf-uva-api/src/main/java/es/uva/inf/tutorias/business/domain/enums/EstadoSolicitud.java
package es.uva.inf.tutorias.business.domain.enums;

public enum EstadoSolicitud {
	PENDIENTE((short) 1, "pendiente"), 
	ACEPTADA((short) 2, "aceptada"),
	RECHAZADA((short) 3, "rechazada"),
	CONFIRMADA((short) 4, "confirmada"),
	CANCELADA((short) 5, "cancelada");
	
	public final short id;
	public final String estado;
	
	EstadoSolicitud(short id, String estado) {
		this.id = id;
		this.estado = estado;
	}
	
	public static EstadoSolicitud getById(short id) {
		for (EstadoSolicitud es : EstadoSolicitud.values()) {
			if (es.id == id) {
				return es;
			}
		}
		
		return EstadoSolicitud.PENDIENTE;
	}
}
